pkg/planner/core/indexmerge_unfinished_path.go

```diff
@@ -350,7 +384,7 @@ func buildIntoAccessPath(
 				if err != nil || !ok || (isIntersection && len(oneAlternative) > 1) {
 					continue
 				}
-				needSelection = len(remainingFilters) > 0 || len(unfinishedPath.idxColHasUsableFilter) > 0
+				needSelection = len(remainingFilters) > 0
```

Comment:
Moved to the check at L211. If you look at it carefully, they are essentially the same.